docs/development/install.rst

```diff
@@ -1,194 +1,267 @@
+In case you need to add a new Python dependency while developing,
+you can use the ``common/dockerfiles/entrypoints/common.sh`` script as shortcut.
+
+This script is run at startup on all the servers (web, celery, builder, proxito) which
+allows you to test your dependency without re-building the whole image.
+To do this, add the ``pip`` command required for your dependency in ``common.sh`` file:
+
+.. code-block:: bash
+
+   # common.sh
+   pip install my-dependency==1.2.3
+
+Once the PR that adds this dependency was merged, you can rebuild the image
+so the dependency is added to the Docker image itself and it's not needed to be installed
+each time the container spins up.
+
+
+Debugging Celery
+~~~~~~~~~~~~~~~~
+
+In order to step into the worker process, you can't use ``pdb`` or ``ipdb``, but
+you can use ``celery.contrib.rdb``:
+
+.. code-block:: python
+
+    from celery.contrib import rdb; rdb.set_trace()
+
+When the breakpoint is hit, the Celery worker will pause on the breakpoint and
+will alert you on STDOUT of a port to connect to. You can open a shell into the container
+with ``inv docker.shell celery`` (or ``build``) and then use ``telnet`` or ``netcat``
+to connect to the debug process port:
+
+.. prompt:: bash
+
+    nc 127.0.0.1 6900
+
+The ``rdb`` debugger is similar to ``pdb``, there is no ``ipdb`` for remote
+debugging currently.
+
+
+Configuring connected accounts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These are optional steps to setup the :doc:`connected accounts </connected-accounts>`
+(GitHub, GitLab, and BitBucket) in your development environment.
+This will allow you to login to your local development instance
+using your GitHub, Bitbucket, or GitLab credentials
+and this makes the process of importing repositories easier.
+
+However, because these services will not be able to connect back to your local development instance,
+:doc:`webhooks </webhooks>` will not function correctly.
+For some services, the webhooks will fail to be added when the repository is imported.
+For others, the webhook will simply fail to connect when there are new commits to the repository.
+
+.. figure:: ../_static/images/development/bitbucket-oauth-setup.png
+    :align: center
+    :figwidth: 80%
+    :target: ../_static/images/development/bitbucket-oauth-setup.png
+
+    Configuring an OAuth consumer for local development on Bitbucket
+
+* Configure the applications on GitHub, Bitbucket, and GitLab.
+  For each of these, the callback URI is ``http://community.dev.readthedocs.io/accounts/<provider>/login/callback/``
+  where ``<provider>`` is one of ``github``, ``gitlab``, or ``bitbucket_oauth2``.
+  When setup, you will be given a "Client ID" (also called an "Application ID" or just "Key") and a "Secret".
+* Take the "Client ID" and "Secret" for each service and enter it in your local Django admin at:
+  ``http://community.dev.readthedocs.io/admin/socialaccount/socialapp/``.
+  Make sure to apply it to the "Site".
+
+
+Core team standards
```

Comment:
I think we can simplify or remove this section as everything that is mentioned is done in docker already.

Reply:
I think it's worth noting this stuff in terms of architecture, but not necessarily in the install doc. I think it's fine tho.

Reply:
I'll keep it here for now. However, I'd be :+1: on moving this into the https://docs.readthedocs.io/en/stable/development/architecture.html together with a nicer graph there.